File: backend/semgrep_rules/tests/python/kisa_sw_40_resource_leak.py
def unsafe(path):
    # ruleid: kisa.sw40.python.file-handle-without-context-review
    f = open(path, "r")
    return f.read()

def safe(path):
    # ok: kisa.sw40.python.file-handle-without-context-review
    with open(path, "r") as f:
        return f.read()

File: backend/semgrep_rules/tests/java/kisa_sw_40_resource_leak.java
import java.io.*;
class kisa_sw_40_resource_leak {
    int unsafe(String path) throws Exception {
        // ruleid: kisa.sw40.java.closeable-without-try-with-resources-review
        InputStream in = new FileInputStream(path);
        return in.read();
    }
    int safe(String path) throws Exception {
        // ok: kisa.sw40.java.closeable-without-try-with-resources-review
        try (InputStream in = new FileInputStream(path)) {
            return in.read();
        }
    }
}
